cpp/benchmarks/CMakeLists.txt

```
@@ -0,0 +1,64 @@
+# =============================================================================
+# Copyright (c) 2025, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+# in compliance with the License. You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under the License
+# is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+# or implied. See the License for the specific language governing permissions and limitations under
+# the License.
+# =============================================================================
+
+#[=======================================================================[.rst:
+kvikio_add_benchmark
+--------------------
+
+Create a KvikIO benchmark.
+
+.. code-block:: cmake
+
+   kvikio_add_benchmark(NAME <name> SOURCES <sources>)
+
+   ``NAME``
+   Benchmark name. Single-value argument.
+
+   ``SOURCES``
+   List of source files for the benchmark. Multi-value argument.
+#]=======================================================================]
+function(kvikio_add_benchmark)
+  cmake_parse_arguments(
+    _KVIKIO # prefix
+    "" # optional
+    "NAME" # single value
+    "SOURCES" # multi-value
+    ${ARGN}
+  )
+
+  if(DEFINED _KVIKIO_UNPARSED_ARGUMENTS)
+    message(FATAL_ERROR "Unknown argument: ${_KVIKIO_UNPARSED_ARGUMENTS}")
+  endif()
+
+  add_executable(${_KVIKIO_NAME} ${_KVIKIO_SOURCES})
+  set_target_properties(${_KVIKIO_NAME} PROPERTIES INSTALL_RPATH "\$ORIGIN/../../../lib")
+
+  get_target_property(KVIKIO_INCLUDES kvikio::kvikio INCLUDE_DIRECTORIES)
+  target_include_directories(${_KVIKIO_NAME} PRIVATE "${KVIKIO_INCLUDES}")
+  target_link_libraries(${_KVIKIO_NAME} PUBLIC benchmark::benchmark kvikio::kvikio)
+
+  if(CMAKE_COMPILER_IS_GNUCXX)
+    set(KVIKIO_CXX_FLAGS "-Wall;-Werror;-Wno-unknown-pragmas")
+    target_compile_options(${_KVIKIO_NAME} PRIVATE "$<$<COMPILE_LANGUAGE:CXX>:${KVIKIO_CXX_FLAGS}>")
+  endif()
```

`CMAKE_COMPILER_IS_GNUCXX` is [deprecated](https://cmake.org/cmake/help/latest/manual/cmake-variables.7.html#deprecated-variables-for-languages). Use the [`CMAKE_CXX_COMPILER_ID`](https://cmake.org/cmake/help/latest/variable/CMAKE_LANG_COMPILER_ID.html) variable instead:

```suggestion
  if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
    set(KVIKIO_CXX_FLAGS "-Wall;-Werror;-Wno-unknown-pragmas")
    target_compile_options(${_KVIKIO_NAME} PRIVATE "$<$<COMPILE_LANGUAGE:CXX>:${KVIKIO_CXX_FLAGS}>")
  endif()
```

---

Thanks! Also updated other places where `CMAKE_COMPILER_IS_GNUCXX` was used.